pkg/network/proxy/proxy.go

```diff
@@ -221,73 +236,65 @@ func (proxy *OsdnProxy) updateEgressNetworkPolicy(policy networkv1.EgressNetwork
 
 	// Add/Update/Delete firewall rules for the namespace
 	if len(firewall) > 0 {
-		if _, ok := proxy.firewall[ns]; !ok {
-			item := &proxyFirewallItem{}
-			item.namespaceFirewalls = make(map[ktypes.UID][]firewallItem)
-			item.activePolicy = nil
-			proxy.firewall[ns] = item
-		}
-		proxy.firewall[ns].namespaceFirewalls[policy.UID] = firewall
-	} else if _, ok := proxy.firewall[ns]; ok {
-		delete(proxy.firewall[ns].namespaceFirewalls, policy.UID)
-		if len(proxy.firewall[ns].namespaceFirewalls) == 0 {
-			delete(proxy.firewall, ns)
-		}
+		ns.firewalls[policy.UID] = firewall
+	} else {
+		delete(ns.firewalls, policy.UID)
 	}
 
 	// Set active policy for the namespace
-	if ref, ok := proxy.firewall[ns]; ok {
-		if len(ref.namespaceFirewalls) == 1 {
-			for uid := range ref.namespaceFirewalls {
-				ref.activePolicy = &uid
-				klog.Infof("Applied firewall egress network policy: %q to namespace: %q", uid, ns)
-			}
-		} else {
-			ref.activePolicy = nil
+	if len(ns.firewalls) == 1 {
+		for uid := range ns.firewalls {
```

Comment:
if len == 1 it must be ns.firewalls[0], no?

Reply:
`ns.firewalls` is a map; if there is only 1 value, we want the key of that value.

Yeah, this code is really weird, but it was already like that before; this code is just being moved around. But yeah, it should probably be rewritten to be less weird.